rust-runtime/aws-smithy-observability/src/provider.rs

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Definitions of high level Telemetry Providers.
+
+use std::sync::Arc;
+
+use crate::{meter::MeterProvider, noop::NoopMeterProvider};
+
+/// A struct to hold the various types of telemetry providers.
+#[non_exhaustive]
+pub struct TelemetryProvider {
+    meter_provider: Box<dyn MeterProvider + Send + Sync>,
+}
+
+impl TelemetryProvider {
+    /// Returns a builder struct for [TelemetryProvider]
+    pub fn builder() -> TelemetryProviderBuilder {
+        TelemetryProviderBuilder {
+            meter_provider: Box::new(NoopMeterProvider),
+        }
+    }
+
+    /// Get the set [MeterProvider]
+    pub fn meter_provider(&self) -> &(dyn MeterProvider + Send + Sync) {
+        self.meter_provider.as_ref()
+    }
+}
+
+impl Default for TelemetryProvider {
+    fn default() -> Self {
+        Self {
+            meter_provider: Box::new(NoopMeterProvider),
+        }
+    }
+}
+
+// If we choose to expand our Telemetry provider and make Logging and Tracing
+// configurable at some point in the future we can do that by adding default
+// logger_provider and tracer_providers based on `tracing` to maintain backwards
+// compatibilty with what we have today.
+/// A builder for [TelemetryProvider].
+#[non_exhaustive]
+pub struct TelemetryProviderBuilder {
+    meter_provider: Box<dyn MeterProvider + Send + Sync>,
+}
+
+impl TelemetryProviderBuilder {
+    /// Set the [MeterProvider].
+    pub fn meter_provider(mut self, meter_provider: Box<dyn MeterProvider + Send + Sync>) -> Self {
```

Comment:
should probably take `impl MeterProvider` or something more flexible.

Reply:
Done in https://github.com/smithy-lang/smithy-rs/pull/3912/commits/0559e8b54fb41bcf04125e2e3cedeec87b24a9f9